#Part 1


###Step 1: Create the dataframe

In [8]:
#a Import pandas and the csv file

#Code cell 1
#import pandas
import pandas as pd
#create a pandas dataframe called "training" from the titanic-train.csv file
training = "/content/titanic_train.csv"

tF = pd.read_csv(training)


In [19]:
#b) Verify the import and take a look at the data.

#Code cell 2
#verify the contents of the training dataframe using the pandas info() method.
#training.

tF.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    int64  
 5   Age          891 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(6), object(4)
memory usage: 83.7+ KB


In [20]:
#Code cell 3
#view the first few rows of the data

tF.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name  Sex   Age  SibSp  Parch  \
0                            Braund, Mr. Owen Harris    0  22.0      1      0   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...    1  38.0      1      0   
2                             Heikkinen, Miss. Laina    1  26.0      0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)    1  35.0      1      0   
4                           Allen, Mr. William Henry    0  35.0      0      0   

             Ticket     Fare Cabin Embarked  
0         A/5 21171   7.2500   NaN        S  
1          PC 17599  71.2833   C85        C  
2  STON/O2. 3101282   7.9250   NaN        S  
3            113803  53.1000  C123        S  
4            373450   8.0500   NaN        S

Most columns have 891 number of values except at "Cabin: 204" and "Embarked: 889".

###Step 2: Prepare the Data for the Decision Tree Model.

In [22]:
#a) Replace string data with numeric labels
#code cell 4
tF["Sex"] = tF["Sex"].apply(lambda toLabel: 0 if toLabel ==
'male' else 1)

In [23]:
#b) Verify that the Gender variable has been changed.

#code cell 5
#view the first few rows of the data again

tF.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name  Sex   Age  SibSp  Parch  \
0                            Braund, Mr. Owen Harris    1  22.0      1      0   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...    1  38.0      1      0   
2                             Heikkinen, Miss. Laina    1  26.0      0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)    1  35.0      1      0   
4                           Allen, Mr. William Henry    1  35.0      0      0   

             Ticket     Fare Cabin Embarked  
0         A/5 21171   7.2500   NaN        S  
1          PC 17599  71.2833   C85        C  
2  STON/O2. 3101282   7.9250   NaN        S  
3            113803  53.1000  C123        S  
4            373450   8.0500   NaN        S

In [24]:
#c) Address Missing Values in the Datase

#code cell 6
tF["Age"].fillna(tF["Age"].mean(), inplace=True)

In [25]:
#d) Verify that the values have been replaced.

#code cell 7
#verify that the missing values for the age variable have been eliminated.

missing_age_values = tF["Age"].isnull().sum()
if missing_age_values == 0:
    print("No missing values in the 'Age' column.")
else:
    print(f"Remaining missing values in the 'Age' column: {missing_age_values}")

No missing values in the 'Age' column.


###Step 3: Train and Score the Decision Tree Model.

In [27]:
#a) Create an array object with the variable that will be the target for the model

#code cell 8
#create the array for the target values
y_target = tF["Survived"].values

In [29]:
#b) Create an array of the values that will be the input for the model.

#code cell 9
columns = ["Fare", "Pclass", "Sex", "Age", "SibSp"]
#create the variable to hold the features that the classifier will use
X_input = tF[list(columns)].values

In [30]:
#c) Create the learned model.
#code cell 10
#import the tree module from the sklearn library
from sklearn import tree
#create clf_train as a decision tree classifier object
clf_train = tree.DecisionTreeClassifier(criterion="entropy", max_depth=3)
#train the model using the fit() method of the decision tree object.
#Supply the method with the input variable X_input and the target variable y_target
clf_train = clf_train.fit(X_input, y_target)


In [32]:
#d) Evaluate the model

#code cell 11
clf_train.score(X_input,y_target)


0.7216610549943884

###Step 4 Visualize the Tree

In [38]:
# a) Create the intermediate file output

#code cell 12
from six import StringIO
with open("/content/titanic_train.csv", 'w') as f:
  f = tree.export_graphviz(clf_train, out_file=f, feature_names=columns)


In [34]:
#b) install Graphviz
!apt-get install graphviz

Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
graphviz is already the newest version (2.42.2-6).
0 upgraded, 0 newly installed, 0 to remove and 35 not upgraded.


In [40]:
#c) Convert the intermediate file to a graphic

#code cell 13
#run the Graphviz dot command to convert the .dot file to .png
!dot -Tpng /content/titanic_train.csv -o /content/titanic_train.csv


In [42]:
#d) Display the image

#code cell 14
#import the Image module from the Ipython.display libary
from IPython.display import Image
#display the decison tree graphic
Image("/content/titanic.png")


FileNotFoundError: No such file or directory: '/content/titanic.png'

FileNotFoundError: No such file or directory: '/content/titanic.png'

<IPython.core.display.Image object>